### PR TITLE
Support Annotations with corrupt /BS-entries

### PR DESCRIPTION
**Pull request by Snuffleupagus**
There's obviously a few things wrong with the Annotations in the referenced PDF document, however parsing of an Annotation shouldn't just break if the /BS-entry isn't a dictionary.

*The PDF document was found here:* https://github.com/py-pdf/pypdf/issues/2332#issuecomment-1846281416

### CODE REVIEW

File: test/unit/annotation_spec.js [hunk 1]
```diff
@@ -35,7 +35,6 @@ import {
 import {
   CMAP_URL,
   createIdFactory,
-  getNodeVersion,
```
Comment (timvandermeij) [inline test/unit/annotation_spec.js:38]: This now seems unused in the project, so let's remove this helper function as well.

File: test/unit/annotation_spec.js [hunk 2]
```diff
@@ -2209,10 +2208,9 @@ describe("annotation", function () {
     });
 
     it("should compress and save text", async function () {
-      if (isNodeJS && getNodeVersion().major >= 20) {
+      if (isNodeJS) {
         pending(
-          "CompressionStream behaves differently in Node.js >= 20, " +
-            "compared to Firefox, Chrome, and Node.js 18."
+          "CompressionStream behaves differently in Node.js, compared to Firefox and Chrome."
```
Comment (timvandermeij) [inline test/unit/annotation_spec.js:2213]: Could you make an issue to track this problem so we can figure out why this behavior changed all of a sudden? I'm fine with disabling it here, but given that this now impacts all Node.js versions it'd be good to have a tracking issue for it.

**Review by timvandermeij — APPROVED** [review]
r=me with these comments addressed. Thanks!

File: test/unit/annotation_spec.js (hunk 2 above)
Comment (Snuffleupagus) [reply test/unit/annotation_spec.js:2213]: Done, please see #17399.

**Pull request merged**